Add's rtype to all constants used in P3 to ensure precision consistency

AaronDonahue opened the pull request:
This commit is non-BFB in that prior code did not enforce precision everywhere
so some of the values may have changed.  B4B consistentcy was checked by forcing
a baseline version of the model to be fully double precision using the intel
fortran compiler option '-fpconstant', then making the changes and comparing.

Thus, when all constants are set to be consistent in terms of precision this commit
is BFB.

Finally, there was a handful of constants that were non-BFB even when precision was
enforced by compiler flag.  These haven't been changed in this commit and will be
addressed in a later commit.  These are clearly marked by comments using the line
"rtype problem".

AaronDonahue committed:
Add's rtype to all constants used in P3 to ensure consistency in precision
calculations.

This commit is non-BFB in that prior code did not enforce precision everywhere
so some of the values may have changed.  B4B consistentcy was checked by forcing
a baseline version of the model to be fully double precision using the intel
fortran compiler option '-fpconstant', then making the changes and comparing.

Thus, when all constants are set to be consistent in terms of precision this commit
is BFB.

Finally, there was a handful of constants that were non-BFB even when precision was
enforced by compiler flag.  These haven't been changed in this commit and will be
addressed in a later commit.  These are clearly marked by comments using the line
"rtype problem".

pressel reviewed (approved):
@AaronDonahue This looks great.

AaronDonahue committed:
Include rtype designation to those variables that didn't match
BFB with the intel compiler option '-fpconstant'.

As mentioned in the previous commit, there were a handful of locations
where the addition of "_rtype" did not match BFB with the original
code using the '-fpconstant' compiler option.  Those were left out
of the previous commit but are fixed here.

NOTE: one inconsistentcy that we tracked down was that the expressions:
thrd = 1./3.
sxth = 1./6.
remained single precision even with the compiler option.  So it makes
sense that the addition of "_rtype" would change these values and
subsequently all values that use thrd and sxth.  This is likely a bug
or issue with "-fpconstant" and not with p3.

AaronDonahue commented:
@pressel , I added in the _rtype for the sections that were not BFB with "-fpconstant" from the previous commit.  I'm going to go ahead and merge without re-review.
